web-app/src/components/BookedLeave/container.js

```diff
@@ -15,8 +15,12 @@ export default Wrapped =>
       this.state = {
         bookedHolidays: [],
       };
+
+      this.cancelHoliday = this.cancelHoliday.bind(this);
+      this.handleCancelHoliday = this.handleCancelHoliday.bind(this);
```

Comment:
Remove the bind lines here. The reason these lines are nessesary is to bind the `this` context to the component's class. Another (and better) way to do this with ES6 is to use the arrow syntax when creating these methods, which keeps the `this` context of the parent (in this case the component class). 

Eg:

```
cancelHoliday = holiday => {
 ...
}
```